### DISCUSSION TITLE
Error in compiling source code with aocc compiler

### DISCUSSION THREAD

**Discussion by ZhengZQ007** [OpenRadioss/OpenRadioss#1720] [Q&A]
## Error in compiling starter
![image](https://github.com/OpenRadioss/OpenRadioss/assets/74033106/fff87f61-0aab-4534-8d47-715724c3aceb)

## Error in compiling engine
![image](https://github.com/OpenRadioss/OpenRadioss/assets/74033106/6a6a2730-a309-44b3-a6d7-8e37fb29fe2c)

**Comment by roguic** [comment]
Hi @ZhengZQ007 ,

I can reproduce the error. 
OpenRadioss uses some macro, and when aocc replaces them, the line length exceeds 132 characters. Aocc is limited to 132 characters (see the -Mextend aocc compiler option), so the compiler complains about it.

We need to work on it.

As a workaround maybe you can use gfortran instead of aocc compiler. If you want to target an AMD processor, you can add some compiler options  (https://gcc.gnu.org/onlinedocs/gcc/x86-Options.html). For example if you want to target an AMD family 19h core, you can use the following command line :
./build_script.sh -arch=linux64_gf -mpi=ompi -addflag="-march=znver4" (engine with openmpi + gfortran + zen4 compiler option)

Hope this help

Best,
Romain

**Reply by ZhengZQ007** [reply]
Many thanks.
